plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -94,7 +109,43 @@ private ConnectorSplit toIcebergSplit(FileScanTask task)
                 task.length(),
                 task.file().fileSizeInBytes(),
                 task.file().format(),
-                ImmutableList.of(),
+                blocklocations(task),
                 getPartitionKeys(task));
     }
+
+    private List<HostAddress> blocklocations(FileScanTask task)
+    {
+        Set<HostAddress> locationSets = new HashSet<>();
+        Path path = new Path(task.file().path().toString());
+        try {
+            FileSystem fs = hdfsEnvironment.getFileSystem(hdfsContext, path);
+            for (BlockLocation b : fs.getFileBlockLocations(path, task.start(), task.length())) {
```

Comment:
This could be very expensive. We need to do it behind a config flag that defaults to disabled.

Reply:
thank you! l will add a config falg